python/pyiceberg/manifest.py

```diff
@@ -208,8 +305,11 @@ def __setattr__(self, name: str, value: Any) -> None:
             value = FileFormat[value]
         super().__setattr__(name, value)
 
-    def __init__(self, *data: Any, **named_data: Any) -> None:
-        super().__init__(*data, **{"struct": DATA_FILE_TYPE, **named_data})
+    def __init__(self, format_version: Literal[1, 2] = 1, *data: Any, **named_data: Any) -> None:
+        super().__init__(
+            *data,
+            **{"struct": DATA_FILE_TYPE if format_version == 1 else data_file_with_partition(StructType(), 2), **named_data},
```

Comment:
The additional `format_version` is added to handle the v2 case, where the field `block_size_in_bytes` should be skipped when written to the manifest file

Reply:
Can you move the creation of the `data_file_with_partition(StructType(), 2)` to a constant as well? I liked how you had the `DATA_FILE_TYPE_V1` and `DATA_FILE_TYPE_V2`

Reply:
Thanks for the suggestion! I made two constants `DATA_FILE_TYPE_V1` and `DATA_FILE_TYPE_V2`. I think in this way we can also simplify the `data_file_with_partition` a llitle bit.